<launch>
   

    <include file="$(find setting_tf2)/launch/tf2_setting.launch">
    </include>
<?ignore
 <!-- Dynamixel Workbench Controller -->
  <include file="$(find dynamixel_workbench_controllers)/launch/position_control.launch">
        <arg name="device_name"        		default="/dev/ttyU2D2"/>
        <arg name="baud_rate"   		default="4500000"/>
	<arg name="profile_velocity"		default="0"/>
	<arg name="profile_acceleration"	default="0"/>
	<arg name="position_p_gain"		default="1700"/>
	<arg name="position_i_gain"		default="500"/>
	<arg name="position_d_gain"		default="300"/>
    </include>  ?>
	<!-- ros_mscl -->
    <include file="$(find microstrain_inertial_driver)/launch/microstrain.launch">
    	</include> 

    <!-- t265 -->
    <include file="$(find realsense2_camera)/launch/rs_t265.launch">
    </include> 
    <!-- rosserial_arduino -->
    <node pkg="rosserial_arduino" type="serial_node.py" name="ros_arduino_PWMs" >
	<param name="port" value="/dev/ttyARDUINO"/>
        <param name="baud" value="57600"/>
    </node>
	<node pkg="pwm_pca9685" type="pca9685_node" name="pwm_generator"/>

    <!-- FAC_MAV_Controller -->
	<!--integration limits-->
	<param name="attitude_integ_limit"	value="10.0"/>
	<param name="attitude_y_integ_limit"      value="3.0"/>
	<param name="altitude_integ_limit"	value="200.0"/>
	<param name="position_integ_limit"	value="3.0"/>

	<!--Center of mass-->
	<param name="x_center_of_mass"		value="-0.02"/>	
	<param name="y_center_of_mass"		value="-0.525"/>	<!-- -0.545 -->
	<param name="z_center_of_mass"		value="-0.085"/>	

	<!--Conventional flight mode-->
    		<!--Conventional Mode attitude_rp gains-->
    		<param name="conv_attitude_r_P_gain"	value="30.0"/>
    		<param name="conv_attitude_r_I_gain"	value="0.75"/>
    		<param name="conv_attitude_r_D_gain"	value="4.0"/>

		<param name="conv_attitude_p_P_gain"   value="15.0"/>
                <param name="conv_attitude_p_I_gain"   value="0.75"/>
                <param name="conv_attitude_p_D_gain"   value="4.0"/>

		<!--Conventional Mode attitude_y gain-->
		<param name="conv_attitude_y_P_gain"	value="30.0"/>
		<!--<param name="conv_attitude_y_I_gain"	value="0"/>-->
		<param name="conv_attitude_y_D_gain"	value="3.0"/>

		<!--Conventional Mode Altitude gains-->
		<param name="conv_altitude_P_gain"	value="11.0"/>
		<param name="conv_altitude_I_gain"	value="5.0"/>
		<param name="conv_altitude_D_gain"	value="6.0"/>

		<!--Conventional Mode Velocity gains-->
		<param name="conv_velocity_P_gain"	value="2.0"/>	
		<param name="conv_velocity_I_gain"	value="1.0"/>	
		<param name="conv_velocity_D_gain"	value="0.3"/>

		<!--Conventional Mode Position gains-->
		<param name="conv_position_P_gain"	value="0.6"/>	
		<param name="conv_position_I_gain"	value="0.1"/>	
		<param name="conv_position_D_gain"	value="0.4"/>
	
	<!--Tilt flight mode-->
		<!--Tilt Mode attitude_r gains-->
		<param name="tilt_attitude_r_P_gain"	value="8.7"/> <!-- 30.0 -->	
		<param name="tilt_attitude_r_I_gain"	value="1.2"/> <!-- 2.4 -->
		<param name="tilt_attitude_r_D_gain"	value="8.4"/> <!-- 19.0 -->
		
		<param name="tilt_attitude_p_P_gain"    value="8.7"/> <!-- 17.4 -->
                <param name="tilt_attitude_p_I_gain"    value="0.2"/> <!-- 2.4 -->
                <param name="tilt_attitude_p_D_gain"    value="8.4"/> <!-- 23.0 --> 

		<!--Tilt Mode attitude_y gains-->
		<param name="tilt_attitude_y_P_gain"	value="15.0"/> <!-- 35.0 -->
		<param name="tilt_attitude_y_I_gain"	value="0.0"/> <!-- 0 -->
		<param name="tilt_attitude_y_D_gain"	value="3.0"/> <!-- 6.0 -->


		<!--Tilt Mode Altitude gains-->
		<param name="tilt_altitude_P_gain"	value="15.0"/> <!--30.0-->
		<param name="tilt_altitude_I_gain"	value="10.0"/> <!--10.0-->
		<param name="tilt_altitude_D_gain"	value="8.0"/> <!--8.0-->

		<!--Tilt Mode Velocity gains-->
		<param name="tilt_velocity_P_gain"	value="1.0"/>	
		<param name="tilt_velocity_I_gain"	value="0.4"/>	
		<param name="tilt_velocity_D_gain"	value="0.4"/>

		<!--Tilt Mode Position gains-->
		<param name="tilt_position_P_gain"	value="2.0"/> <!--3.0 -->	
		<param name="tilt_position_I_gain"	value="0.002"/>	
		<param name="tilt_position_D_gain"	value="2.0"/>	<!--4.0 -->

    <node name="cad_uav_main_node"  pkg="FAC_MAV" type="cad_uav_main" output="screen"/>

	<node pkg="data_logging" name="data_logging_node" type="data_logging"/>
    
	<!-- Rosbag -->
  <node pkg="rosbag" type="record" name="record" output="screen" args="-o /home/paletrone/catkin_ws/src/FAC_MAV_paletrone/FAC_MAV/bag/ /data_log" /><!-- /angle /desired_angle /Forces /t265_pos /desired_altitude"/> --> 
</launch>  
